Automattic/woocommerce-payments
# Refactor `loadStripe` for Express Checkouts

Closes #9685 

This refactor originated from a comment (pcrvl0-1sv-p2#comment-743) on a P2 I published. It doesn’t introduce any new functionality. The changes are solely intended to prevent reintroducing the same bug (pcrvl0-1sv-p2) that https://github.com/Automattic/woocommerce-payments/pull/9523 fixed it.

#### Changes proposed in this Pull Request

Since loadStripe was introduced for Express Checkouts, the `forceAccountRequest` parameter is unnecessary, as it should always use the merchant’s connected account with Stripe elements.

Therefore, the `forceAccountRequest` parameter is being removed, and the function renamed to clearly indicate it’s for Express Checkouts.

All instances of `loadStripe` has been updated to use the new `loadStripeForExpressCheckout`





#### Testing instructions

> [!TIP]
> To thoroughly test these changes, I created a plugin .zip file using `npm run build:release` and applied it to a fresh Jurassic Ninja site. This ensures the domain hadn’t been previously registered.

> [!NOTE]  
> You need to use Ngrok or Jurassic Tube to test Apple Pay.

* Switch to this branch.
* Add a product to the cart
* Go to the cart or the checkout page.
* Pay using Google Pay or Apple Pay.
* Order should be processed without issues.



-------------------


**Post merge**
